package.json

```diff
@@ -122,5 +128,13 @@
   },
   "optionalDependencies": {
     "gulp-appdmg": "^1.0.3"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.(js|vue)": "eslint --cache --fix"
```

Comment:
I am not a big fan of committing automatically generated / modified code to into a repository - in my opinion this is of little value, and contributors should instead own their changes / nip bad formatting at the source by fixing the auto-formatting in their tooling.
So can I propose that, instead of auto-formatting code as part of every commit (which is un-intuitive as well), that we simply err out in this case? Users who want to use auto-formatting can always do so with a manual command invocation.
Also, the linter should also be run as part of the tests, so that bad code will be found as part of CI/CD - as I explained before, a considerable fraction of contributors are not running a full local build chain, and will not be able to use this locally.

Reply:
What's the preference for CI then? Travis or GitHub action?

Reply:
Travis isn't working for us any more, since they have introduced a fixed limit on machine hours for open source projects, so it is Azure (currently used) or GitHub Action.
I think the simplest way is to simply add the linter checks to the `test` yarn task, which is already run by CI (and should be run by developers as well to make sure that their changes aren't breaking any tests.